### DISCUSSION TITLE
unable to setup it locally getting error while running yarn help please ?

### DISCUSSION THREAD

**Discussion by sahilchauhan123** [RocketChat/Rocket.Chat#32576] [Q&A] [answered]
YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 1s 944ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ yocto-queue@npm:0.1.0 can't be found in the cache and will be fetch
➤ YN0013: │ yocto-queue@npm:1.0.0 can't be found in the cache and will be fetch
➤ YN0013: │ zip-stream@npm:2.1.3 can't be found in the cache and will be fetche
➤ YN0013: │ zustand@npm:4.3.8 can't be found in the cache and will be fetched f
➤ YN0013: │ zwitch@npm:1.0.5 can't be found in the cache and will be fetched fr
➤ YN0066: │ typescript@patch:typescript@npm%3A5.3.3#~builtin<compat/typescript>::version=5.3.3&hash=85af82: Cannot apply hunk #1
➤ YN0000: └ Completed in 2m 56s
➤ YN0000: ┌ Link step
➤ YN0007: │ @kaciras/deasync@npm:1.0.3 must be built because it never has been before or the last one failed
➤ YN0007: │ bcrypt@npm:5.0.1 must be built because it never has been before or the last one failed
➤ YN0007: │ fibers@npm:5.0.3 must be built because it never has been before or the last one failed
➤ YN0007: │ highlight.js@npm:9.18.5 must be built because it never has been before or the last one failed
➤ YN0007: │ isolated-vm@npm:4.4.2 must be built because it never has been before or the last one failed
➤ YN0007: │ sharp@npm:0.32.6 must be built because it never has been before or the last one failed
➤ YN0007: │ sodium-native@npm:3.3.0 must be built because it never has been before or the last one failed
➤ YN0007: │ @swc/core@npm:1.3.95 [86b19] must be built because it never has been before or the last one failed
➤ YN0007: │ @rocket.chat/forked-matrix-sdk-crypto-nodejs@npm:0.1.0-beta.13 must be built because it never has been before or the last one failed
➤ YN0007: │ @napi-rs/pinyin@npm:1.7.0 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js@npm:3.21.1 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js-pure@npm:3.21.1 must be built because it never has been before or the last one failed
➤ YN0007: │ gc-stats@npm:1.4.0 must be built because it never has been before or the last one failed
➤ YN0007: │ husky@npm:4.3.8 must be built because it never has been before or the last one failed
➤ YN0007: │ event-loop-stats@npm:1.4.1 must be built because it never has been before or the last one failed
➤ YN0007: │ esbuild@npm:0.17.19 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js@npm:2.6.12 must be built because it never has been before or the last one failed
➤ YN0007: │ gifsicle@npm:5.3.0 must be built because it never has been before or the last one failed
➤ YN0007: │ mozjpeg@npm:7.1.1 must be built because it never has been before or the last one failed
➤ YN0007: │ optipng-bin@npm:7.0.1 must be built because it never has been before or the last one failed
➤ YN0007: │ pngquant-bin@npm:6.0.1 must be built because it never has been before or the last one failed
➤ YN0007: │ cwebp-bin@npm:7.0.1 must be built because it never has been before or the last one failed
➤ YN0009: │ isolated-vm@npm:4.4.2 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-d3147482/build.log)
➤ YN0009: │ event-loop-stats@npm:1.4.1 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-4246ed28/build.log)
➤ YN0009: │ gc-stats@npm:1.4.0 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-325a5b8e/build.log)
➤ YN0009: │ gc-stats@npm:1.4.0 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-390bfacf/build.log)
➤ YN0000: └ Completed in 1m 9s
➤ YN0000: Failed with errors in 4m 7s

**Comment by tassoevan** [comment, marked as answer]
Check the build logs generated for the last dependencies listed in the output (`isolated-vm`, `event-loop-stats`, `gc-stats`). There is a good chance you're missing some dependencies to compile native packages and/or using a incorrect Node version.

**Reply by sahilchauhan123** [reply]
yes i was due to incorrect node version

**Reply by Nikhilh26** [reply]
Hey, I am using windows I am getting the same error even with using the specified node version

**Reply by sammamama** [reply]
what node version was successful. Cause I am trying with 22.11.0. And it's failing. I've also tried with 20.18.1

**Reply by Nikhilh26** [reply]
It worked with 20.18.1 only . Are you setting up in windows?

**Comment by B8rislav** [comment]
If you are on this page now, using linux and getting the same error. If you've got `ModuleNotFoundError: No module named 'distutils'` in build logs, you can use answer from [here](https://stackoverflow.com/questions/77251296/distutils-not-found-when-running-npm-install).

Hope it will save you some time

**Comment by reetp** [comment]
And closing
